docs/how-to/websites-on-ipfs/gateway-redirects.md

```diff
@@ -0,0 +1,91 @@
+---
+title: Gateway redirects
+description: What gateway redirects are and how to use them with a website on IPFS.
+---
+
+# THIS IS A DRAFT. DO NOT MERGE.
+
+# To do:
+1. How to redirect old URL to a new place (301 and 302 redirects)
+1. How to use it for PWA/SPA hosting (catch-all 200) 
+1. How to use it to provide custom `404 not found` pages (superceding what `ipfs-404.html` does - ideally not mention old way)
+
+# Gateway Redirects
+
+Gateway Redirects provide support for URL redirects and rewrites for websites hosted on Subdomain or DNSLink gateways. This feature enables support for single-page applications, and avoids link rot when moving to IPFS-backed hosting.
+
+Using Gateway Redirects, you can change the appearance of a URL, change where content is located without breaking existing links, redirect invalid URLs to a custom 404 page, and enable URL rewriting.
+
+# Supported HTTP status codes
+
+* `200` - OK (redirect will be treated as a rewrite, returning OK without changing the URL shown in the browser).
+* `301` - Permanent redirect (the default status).
+* `302` - Found (commonly used for temporary redirects).
+* `303` - See other (replaces PUT and POST with GET).
+* `307` - Temporary redirect (preserves the body and HTTP method of the original request).
+* `308` - Permanent redirect (preserves the body and HTTP method of the original request).
```

Comment:
Due to legacy reasons, Kubo may not support these initially. 
Most people care about 301 and 302, so we may skip other ones in user docs for now:
```suggestion
<!-- TODO: uncomment when supported by Kubo
* `303` - See other (replaces PUT and POST with GET).
* `307` - Temporary redirect (preserves the body and HTTP method of the original request).
* `308` - Permanent redirect (preserves the body and HTTP method of the original request).
-->
```

Reply:
Removing `303` `307` and `308` from doc for now, can be added back in when they get support.